go/cmd/vtctldclient/internal/command/reparents.go

```diff
@@ -261,6 +263,7 @@ func commandTabletExternallyReparented(cmd *cobra.Command, args []string) error
 func init() {
 	EmergencyReparentShard.Flags().DurationVar(&emergencyReparentShardOptions.WaitReplicasTimeout, "wait-replicas-timeout", *topo.RemoteOperationTimeout, "Time to wait for replicas to catch up in reparenting.")
 	EmergencyReparentShard.Flags().StringVar(&emergencyReparentShardOptions.NewPrimaryAliasStr, "new-primary", "", "Alias of a tablet that should be the new primary. If not specified, the vtctld will select the best candidate to promote.")
+	EmergencyReparentShard.Flags().BoolVar(&emergencyReparentShardOptions.PreventCrossCellPromotion, "prevent-cross-cell-promotion", false, "Only promotes a new primary from the same cell as the previous primary")
```

Comment:
Actually, me and @harshit-gangal thought about this too, but we ended up choosing `PreventCrossCellPromotion` with false as default to `EnableCrossCellPromotion` with true as default because of upgrade, downgrade considerations. If a user upgrades their vtctld server but does not upgrade their vtctl client, then they cannot pass the newly created flag, so it would not be in the proto message. which means that it will be set to `false` by default as specified in https://developers.google.com/protocol-buffers/docs/proto3#default. So, we wanted to keep a flag which had a default of false instead of one which defaulted to true.

Reply:
Ah this makes sense.